src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -498,50 +545,83 @@ private async Task<bool> OnSetVariableValue(MessageId id, int scopeId, string va
             var varToSetValue = varIds.FirstOrDefault(v => v.Name == varName);
             if (varToSetValue == null)
                 return false;
-            Result res = await SendMonoCommand(id, MonoCommands.SetVariableValue(scopeId, varToSetValue.Index, varName, varValue["value"].Value<string>()), token);
-            if (res.IsOk)
+            var res = await sdbHelper.SetVariableValue(id, ctx.ThreadId, scopeId, varToSetValue.Index, varValue["value"].Value<string>(), token);
+            if (res)
                 SendResponse(id, Result.Ok(new JObject()), token);
             else
                 SendResponse(id, Result.Err($"Unable to set '{varValue["value"].Value<string>()}' to variable '{varName}'"), token);
             return true;
         }
 
-        private async Task<Result> RuntimeGetProperties(MessageId id, DotnetObjectId objectId, JToken args, CancellationToken token)
+        internal async Task<Result> RuntimeGetProperties(SessionId id, DotnetObjectId objectId, JToken args, CancellationToken token)
         {
-            if (objectId.Scheme == "scope")
+            var accessorPropertiesOnly = false;
+            var ownProperties = false;
+            if (args != null)
             {
-                return await GetScopeProperties(id, int.Parse(objectId.Value), token);
+                if (args["accessorPropertiesOnly"] != null)
+                    accessorPropertiesOnly = args["accessorPropertiesOnly"].Value<bool>();
+                if (args["ownProperties"] != null)
+                    ownProperties = args["ownProperties"].Value<bool>();
             }
-
-            Result res = await SendMonoCommand(id, MonoCommands.GetDetails(objectId, args), token);
-            if (res.IsErr)
-                return res;
-
-            if (objectId.Scheme == "cfo_res")
-            {
-                // Runtime.callFunctionOn result object
-                string value_json_str = res.Value["result"]?["value"]?["__value_as_json_string__"]?.Value<string>();
-                if (value_json_str != null)
+            //Console.WriteLine($"RuntimeGetProperties - {args}");
```

Comment:
Something like
```csharp
            //Console.WriteLine($"RuntimeGetProperties - {args}");
            try {
                switch (objectId.Scheme)
                {
                    case "scope":
                        return await GetScopeProperties(id, int.Parse(objectId.Value), token);
                    case "valuetype":
                        return Result.OkFromObject(new { result = await sdbHelper.GetValueTypeValues(id, int.Parse(objectId.Value), accessorPropertiesOnly, token)});
                    case "array":
                        return Result.OkFromObject(new { result = await sdbHelper.GetArrayValues(id, int.Parse(objectId.Value), token)});
                    case "object":
                        return Result.OkFromObject(new { result = await sdbHelper.GetObjectValues(id, int.Parse(objectId.Value), true, false, accessorPropertiesOnly, ownProperties, token) });
                    case "pointer":
                        return Result.OkFromObject(new { result = await sdbHelper.GetPointerContent(id, int.Parse(objectId.Value), token)});
                    case "cfo_res":
                    {
                        Result res = await SendMonoCommand(id, MonoCommands.GetDetails(int.Parse(objectId.Value), args), token);
                        // Runtime.callFunctionOn result object
                        string value_json_str = res.Value["result"]?["value"]?["__value_as_json_string__"]?.Value<string>();
                        return Result.OkFromObject(new { result = value_json_str != null ? (object) JArray.Parse(value_json_str) : new {} });
                    }
                    default:
                        return Result.Err($"Unable to RuntimeGetProperties '{objectId}'");
                }
            }
            catch (Exception e) {
                return Result.Err($"Unable to RuntimeGetProperties '{objectId}' - {e}");
            }
```

here instead of the repeated ifs?

Reply:
Done.